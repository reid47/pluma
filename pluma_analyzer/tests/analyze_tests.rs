#[macro_use]
mod macros;

test_analyze! {
  undefined_identifier_simple (false): r#"
    |wat
  "#,

  defined_identifier_simple (true): r#"
    |let fine = 47
    |
    |fine
  "#,

  intrinsic_def_binary_op_valid (true): r#"
    |intrinsic_type Int
    |intrinsic_def Int + Int -> Int
    |
    |let result = 3 + 4
  "#,

  intrinsic_def_binary_op_invalid (false): r#"
    |intrinsic_type Int
    |intrinsic_def Int + Int -> Int
    |
    |let result = 3 + "yikes"
  "#,
}
