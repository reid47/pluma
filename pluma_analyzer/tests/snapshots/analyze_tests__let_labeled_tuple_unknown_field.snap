---
source: pluma_analyzer/tests/analyze_tests.rs
expression: "intrinsic_type Int\nintrinsic_type String\n\nlet tup = (name: \"Reid\", age: 26)\nlet (age: age2, wat: name2) = tup"
---

=== Source ===
intrinsic_type Int
intrinsic_type String

let tup = (name: "Reid", age: 26)
let (age: age2, wat: name2) = tup

=== Diagnostics ===
[
    Diagnostic {
        kind: Error,
        message: "Cannot destructure unknown field \'wat\' in pattern. Value has type (name: String, age: Int).",
        pos: Some(
            (
                92,
                95,
            ),
        ),
        module_name: None,
        module_path: None,
    },
]

